src/utils/axis-tick-label-ellipsis.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * A util functions for handling long tick labels
+ */
+
+export const axisTickLableEllipsis = (
+  categoryOrder: string[],
+  maxIndependentTickLabelLength: number
+) => {
+  // make array for tick label with ellipsis
+  const categoryOrderEllipsis = categoryOrder.map((element) => {
+    return (element || '').length > maxIndependentTickLabelLength
+      ? (element || '').substring(0, maxIndependentTickLabelLength) + '...'
+      : element;
+  });
+
+  // identify duplicate element and duplicate indices in the array
+  const duplicateIndexValue = getDuplicates(categoryOrderEllipsis);
+
+  // looping object to map data's label and label with ellipsis
+  Object.entries(duplicateIndexValue).forEach((entry: any) => {
+    const [key, value] = entry;
+    // starting from i = 1 so that the first item is not changed
+    for (let i = 1; i < (value as number[]).length; i++) {
+      // add incremental space(s) for duplicates
+      categoryOrderEllipsis[value[i]] =
+        categoryOrderEllipsis[value[i]] + ' '.repeat(i);
+    }
+  });
+
+  return categoryOrderEllipsis;
+};
+
+/**
+ *  Returning an object comprised of of duplicate(s) and corresponding index array
+ *    Input: an array that has duplicate elements
+ *    Output: an object, e.g.,
+ *      { 'duplicate element 1': [0, 1, 3], 'duplicate element 2': [5, 7], ... }
+ */
+const getDuplicates = (arr: string[]) => {
+  var duplicates: any = {};
```

Comment:
For this declaration, I would:

1. Use `const` instead of `var`.
2. Add an explicit typing of `Record<string, number[]>`.

```suggestion
  const duplicates: Record<string, number[]> = {};
```

Reply:
@jtlong3rd thank you for your review and suggestion! 👍 Oops moments as I didn't carefully check the function. for typing. I have complied with your suggestion and updated this PR.